connector/connect/client/jvm/pom.xml

```diff
@@ -105,11 +105,16 @@
           <artifactSet>
             <includes>
               <include>com.google.guava:*</include>
+              <include>io.grpc:*:</include>
```

Comment:
@zhenlineo When I run 

`mvn clean install -DskipTests -pl connector/connect/client/jvm`

I can see the shaded logs of grpc, and the server side also writes the same

<img width="857" alt="image" src="https://user-images.githubusercontent.com/1475305/215918872-0d5ad3f1-da33-4851-8f74-78e153caacc0.png">

https://github.com/apache/spark/blob/4d37e7816ce83dc5928b0d42a536304cb49eb0a3/connector/connect/server/pom.xml#L244

Reply:
But let me test your suggestion first

Reply:
@zhenlineo double check this change, it only shaded grpc but  not relocation.

Reply:
More, I checked the packaging results of the `connect/server` module, It is no longer correct, many packages have not been relocated to `org.sparkproject` package. 

I reset the code to [[SPARK-40677][CONNECT][FOLLOWUP] Refactor shade `relocation/rename` rules](https://github.com/apache/spark/pull/38162) and re-checked the packaging results of the `connect/server` module, It's right on that snapshot, all shaded packages are relocated to `org.sparkproject` package.

I'm not sure what changed the packaging results of `server` module, but I think we need to investigate and fix the maven shaded  configuration of the `server` modules to ensure that all the shaded packages are relocated to `org.sparkproject` package, I will create a new Jira to tracking this.